src/extension/ExtensionContextHandler.ts

```diff
@@ -58,5 +61,19 @@ export class ExtensionContextHandler {
         });
         this.context.subscriptions.push(integrationsTreeView);
         this.context.subscriptions.push(vscode.commands.registerCommand('kaoto.integrations.refresh', () => integrationsProvider.refresh()));
+        this.registerNewCamelYamlFiles();
+    }
+
+    private registerNewCamelYamlFiles() {
+        // register commands for new Camel files creation using YAML DSL - Routes, Kamelets, Pipes
+        this.context.subscriptions.push(vscode.commands.registerCommand(NewCamelRouteCommand.ID_COMMAND_CAMEL_ROUTE_YAML, async () => {
+            await new NewCamelRouteCommand('YAML').create();
+        }));
+        this.context.subscriptions.push(vscode.commands.registerCommand(NewCamelKameletCommand.ID_COMMAND_CAMEL_KAMELET_YAML, async () => {
+            await new NewCamelKameletCommand('YAML').create();
+        }));
+        this.context.subscriptions.push(vscode.commands.registerCommand(NewCamelPipeCommand.ID_COMMAND_CAMEL_PIPE_YAML, async () => {
+            await new NewCamelPipeCommand('YAML').create();
+        }));
```

Comment:
for another iteration, provide telemtry https://issues.redhat.com/browse/FUSETOOLS2-2552